### PR TITLE
Migrate Accordion web component to now render using LitElement

### PR DESCRIPTION
**Pull request by sghoweri**
## Jira
http://vjira2:8080/browse/BDS-2125

## Summary
Migrates our existing Accordion web component to now render using LitElement. Addresses catastrophic SSR hydration issues that can sometimes occur when asynchronously loading the component's JS / being lazily rendered (for example, [as seen here](https://boltdesignsystem-pubfgtblg.now.sh/pattern-lab/patterns/02-components-accordion-40-accordion-content-variations/02-components-accordion-40-accordion-content-variations.html)).

## Details
As far as the Accordion component goes, this is more or less a 1:1 upgrade with only minor logic tweaks required, plus a new interaction part of the existing Jest test added as well.

Note: As an added bonus, updating the `spacing` prop (when an accordion item is already open -- ex. going from small to large spacing, or vice-versa) now correctly recalculates the height of the open panel which doesn't happen in the current version...

## How to test
- Confirm that with these upgrades, the Accordion component renders correctly (despite it now being lazily loaded on this branch + renders using `<ssr-keep>`.

- Also confirm that contextual props (ex. `single`, `spacing`, etc) also trigger the Accordion Items to re-render as expected.

### CODE REVIEW

**Comment by remydenton** [comment]
I unfortunately don't have enough knowledge of the specifics of litHtml and litElement to give a meaningful code review of this one.  So I'm not going to approve since I think that needs someone else, but what I do understand seems reasonable, so I'll give it a 👍 instead.

> This PR also splits up our @bolt/element decorators into separate files
It looks to me like this part of the PR description is not longer true (splitting up the decorators was handled in some other PR and then merged back into this one).

File: packages/components/bolt-accordion/src/accordion.js [hunk 1]
```diff
@@ -87,19 +110,6 @@ class BoltAccordion extends withContext(withLitHtml) {
     this.addEventListener('bolt:layout-size-changed', this.handleLayoutChanged);
   }
 
-  getModifiedSchema(schema) {
```
Comment (danielamorse) [inline packages/components/bolt-accordion/src/accordion.js:90]: @sghoweri how/where do we handle this now? We still want to remove props like `items`, `content` from the schema before validating.

Comment (sghoweri) [reply packages/components/bolt-accordion/src/accordion.js:90]: I'd recommend coming up with a [declarative] way to exclude certain props within the component's schema that we can then filter out when processing (a flag or something).

In the interim, physically modifying the schema data (once imported into the JS file) should be sufficient.

For example:
```
import schema from '../accordion.schema';

delete schema.properties.items // or whatever else we want to exclude

...
class BoltAccordion extends withContext(BoltElement) {
  static schema = schema;
}
```

Make sense?

**Comment by sghoweri** [comment]
PR was released with v2.23.0

**Pull request merged**